kpatch-build/create-diff-object.c

```diff
@@ -2381,8 +2445,6 @@ int main(int argc, char *argv[])
 	kpatch_mark_ignored_sections(kelf_patched);
 	kpatch_compare_correlated_elements(kelf_patched);
 	kpatch_check_fentry_calls(kelf_patched);
-	kpatch_elf_teardown(kelf_base);
-	kpatch_elf_free(kelf_base);
```

Comment:
This base elf teardown code is at this location for a good reason: by ensuring the 'base' structures aren't used in the rest of the file, it keeps things nicely separated, which helps keep the code simpler and makes it easier to flush out bugs.  Which gives me an idea for how to rearrange the code in this patch.

Since the local symbols are part of the lookup table, maybe we could pass kelf_base->symbols to lookup_open() at the beginning of the program, after calling kpatch_elf_open().  lookup_open() could find the local syms and setup table->local_syms.  If it can't find a unique matching FILE+locals combination, it could error out.

(BTW, I wrote a python script to find such cases, and looked for duplicate FILE+locals combinations in the vmlinux symbol table for an allyesconfig kernel, and I only found 4 instances in an allyesconfig kernel, and they were in drivers that are normally not compiled into vmlinux.)

Then, later, lookup_local_symbol() could do its lookup based on table->local_syms.  Then there wouldn't be the concept of an "ambiguous" symbol, and so we would no longer need kpatch_resolve_ambiguous_symbol().

I think that would simplify this code quite a bit.  Thoughts?